File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/managers/entity/ITimerEntityManager.java
package com.dkellycollins.triggerfinger.managers.entity;

import com.dkellycollins.triggerfinger.data.entity.ITimer;

/**
 * Created by Devin on 9/5/2015.
 */
public interface ITimerEntityManager {

    Iterable<ITimer> retrieve();
    ITimer retrieve(int id);

    int create(int setTime, boolean isRunning);

    void update(int id, int currentTime);
    void update(int id, int currentTime, boolean isRunning);
    void reset(int id);
    void start(int id);
    void pause(int id);

    void delete(int id);

}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/managers/entity/ITimerEntityManager.kt
package com.dkellycollins.triggerfinger.managers.entity

import com.dkellycollins.triggerfinger.data.entity.ITimer

/**
 * Created by Devin on 9/5/2015.
 */
interface ITimerEntityManager {

    fun retrieve(): Iterable<ITimer>
    fun retrieve(id: Int): ITimer

    fun create(setTime: Int, isRunning: Boolean): Int

    fun update(id: Int, currentTime: Int)
    fun update(id: Int, currentTime: Int, isRunning: Boolean)
    fun reset(id: Int)
    fun start(id: Int)
    fun pause(id: Int)

    fun delete(id: Int)

}
